cmd/gomander/thirdpartyserver/handlers.go

```diff
@@ -0,0 +1,152 @@
+package thirdpartyserver
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"gomander/internal/command/domain"
+	domain2 "gomander/internal/commandgroup/domain"
+	"gomander/internal/helpers/array"
+)
+
+func (s *ThirdPartyIntegrationsServer) handleDiscovery(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, err := w.Write([]byte(`{"app": "Gomander"}`))
+	if err != nil {
+		println("Error writing response:", err.Error())
+	}
+}
+
+func (s *ThirdPartyIntegrationsServer) handleGetCommands(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	commands, err := s.useCases.GetCommands.Execute()
+	if err != nil {
+		http.Error(w, "Failed to get commands", http.StatusInternalServerError)
+		return
+	}
+	runningCommandsIds := s.useCases.GetRunningCommandIds.Execute()
+
+	mappedCommands := array.Map(commands, func(cmd domain.Command) map[string]interface{} {
+		status := "stopped"
+		if array.Contains(runningCommandsIds, cmd.Id) {
+			status = "running"
+		}
+
+		return map[string]interface{}{
+			"id":     cmd.Id,
+			"name":   cmd.Name,
+			"status": status,
+		}
+	})
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(mappedCommands)
+	if err != nil {
+		http.Error(w, "Failed to encode commands", http.StatusInternalServerError)
+	}
+
+}
+
+func (s *ThirdPartyIntegrationsServer) handleRunCommand(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Extract command ID from URL
+	id := r.PathValue("id")
```

Comment:
What do you think about adding some validation for the path parameters? Right now if someone passes an empty ID it might cause issues.


```suggestion
	id := r.PathValue("id")
	if id == "" {
      http.Error(w, "Command ID is required", http.StatusBadRequest)
      return
  }
```

Reply:
@noel-lopez I included this validations at first, but then I tried and, if I'm not wrong, it is not possible to pass and empty value, if you do that, the handler will not match. If you are able to reproduce it, please explain how so I can test it.